# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGTransformableElement.cpp
# Commit: b848ca90888e
# Full Hash: b848ca90888ee8e446f3d54b65c3ef68c71a1451
# Author: longsonr <longsonr@gmail.com>
# Date: 2019-06-20 22:06:31
# Regressor Bug: 1551030
# File Overlap Count: 2
# Description:
#   Bug 1551030 - Merge all SVGRect classes r=dholbert
# ==============================================================================

diff -r 20956a783e08 -r b848ca90888e dom/svg/SVGTransformableElement.cpp
--- a/dom/svg/SVGTransformableElement.cpp	Thu Jun 20 14:19:57 2019 +0300
+++ b/dom/svg/SVGTransformableElement.cpp	Thu Jun 20 07:03:54 2019 -0700
@@ -156,7 +156,7 @@
   return SVGContentUtils::GetOuterSVGElement(this);
 }
 
-already_AddRefed<SVGIRect> SVGTransformableElement::GetBBox(
+already_AddRefed<SVGRect> SVGTransformableElement::GetBBox(
     const SVGBoundingBoxOptions& aOptions, ErrorResult& rv) {
   nsIFrame* frame = GetPrimaryFrame(FlushType::Layout);
 
@@ -199,14 +199,14 @@
     rec.x += float(text->GetPosition().x) / AppUnitsPerCSSPixel();
     rec.y += float(text->GetPosition().y) / AppUnitsPerCSSPixel();
 
-    return NS_NewSVGRect(this, ToRect(rec));
+    return do_AddRef(new SVGRect(this, ToRect(rec)));
   }
 
   if (!NS_SVGNewGetBBoxEnabled()) {
-    return NS_NewSVGRect(
+    return do_AddRef(new SVGRect(
         this, ToRect(nsSVGUtils::GetBBox(
                   frame, nsSVGUtils::eBBoxIncludeFillGeometry |
-                             nsSVGUtils::eUseUserSpaceOfUseElement)));
+                             nsSVGUtils::eUseUserSpaceOfUseElement))));
   }
   uint32_t flags = 0;
   if (aOptions.mFill) {
@@ -222,14 +222,15 @@
     flags |= nsSVGUtils::eBBoxIncludeClipped;
   }
   if (flags == 0) {
-    return NS_NewSVGRect(this, 0, 0, 0, 0);
+    return do_AddRef(new SVGRect(this, gfx::Rect()));
   }
   if (flags == nsSVGUtils::eBBoxIncludeMarkers ||
       flags == nsSVGUtils::eBBoxIncludeClipped) {
     flags |= nsSVGUtils::eBBoxIncludeFill;
   }
   flags |= nsSVGUtils::eUseUserSpaceOfUseElement;
-  return NS_NewSVGRect(this, ToRect(nsSVGUtils::GetBBox(frame, flags)));
+  return do_AddRef(
+      new SVGRect(this, ToRect(nsSVGUtils::GetBBox(frame, flags))));
 }
 
 already_AddRefed<SVGMatrix> SVGTransformableElement::GetCTM() {